### PR TITLE
Interceptor and Special Response ERT redesign

### PR DESCRIPTION
**Pull request by SomrNetwork**
## About The Pull Request
As stated in the title, Redesigned the SpecOps ERT with new clothes and armor. Also adds a medic to their team.
Interceptor body armor or Crasher body armor is the new armor made for the SpecOps and Marines!
Crasher body armor also comes with built in pockets (as all armors should)

image

image
## Why It's Good For The Game
Renews ERTs and adds some potential for new ERTs (HECU coming soon if this gets merged)
New body armor style
## Changelog
:cl:
add: Crasher body armor in marine_armor.dmi
add: New swat mask for SpecOps
add: SpecOps Medic
/ 🆑

### CODE REVIEW

**Comment by thatguythere03** [comment]
fitzshollsealofquality
My name is Fitz Sholl and I approve of this PR.

**Comment by Lumipharon** [comment]
Kill this man, he removed the pistol holsters.

**Comment by thatguythere03** [comment]
> Kill this man, he removed the pistol holsters.

Killed the holsters, added the sexy new body armor. Law of equivalent exchange.

File: code/modules/clothing/suits/marine_armor.dm [hunk 1]
```diff
@@ -697,3 +697,167 @@
...
+	slowdown = SLOWDOWN_ARMOR_LIGHT
+	flags_armor_protection = CHEST|GROIN
+	icon = 'icons/mob/clothing/suits/marine_armor.dmi'
+	item_icons = list(
+		slot_wear_suit_str = 'icons/mob/clothing/suits/marine_armor.dmi'
+	)
+	icon_state = "m_ballistic_vest"
+	flags_armor_features = NONE
+
+	allowed = list(
```
Comment (Mrrpip) [inline code/modules/clothing/suits/marine_armor.dm:779]: Wouldn't this already be inherited from the base version of /obj/item/clothing/suit/storage/marine/ballistic ?

**Review by Kuro020 — CHANGES_REQUESTED** [review]
here comes the kuro 1984

the mask is just the swat gas mask but edited, which is guh
the clothing is a mix of turtle neck and marine normal fatigue sprites, which is guh

also we're not allowed to used red crosses because reasons idk so that gotta go

File: code/modules/clothing/suits/marine_armor.dm [hunk 2]
```diff
@@ -697,3 +697,107 @@
...
+	desc = "A reused design of a old body armor system from the 21st century."
+	soft_armor = list(MELEE = 30, BULLET = 50, LASER = 20, ENERGY = 25, BOMB = 30, BIO = 5, FIRE = 25, ACID = 30)
+	slowdown = SLOWDOWN_ARMOR_LIGHT
+	flags_armor_protection = CHEST|GROIN
+	icon = 'icons/mob/clothing/suits/marine_armor.dmi'
+	item_icons = list(
+		slot_wear_suit_str = 'icons/mob/clothing/suits/marine_armor.dmi'
+	)
+	icon_state = "s_ballistic_vest"
+	flags_armor_features = NONE
```
Comment (Lumipharon) [inline code/modules/clothing/suits/marine_armor.dm:803]: We have colour variants for a reason, stop making separate item types.

File: code/datums/jobs/job/special_forces.dm [hunk 3]
```diff
@@ -42,11 +42,6 @@
 	H.equip_to_slot_or_del(new /obj/item/ammo_magazine/smg/m25/ap, SLOT_IN_BELT)
 	H.equip_to_slot_or_del(new /obj/item/ammo_magazine/smg/m25/ap, SLOT_IN_BELT)
 
-	H.equip_to_slot_or_del(new /obj/item/ammo_magazine/pistol/g22, SLOT_IN_ACCESSORY)
-	H.equip_to_slot_or_del(new /obj/item/ammo_magazine/pistol/g22, SLOT_IN_ACCESSORY)
-	H.equip_to_slot_or_del(new /obj/item/ammo_magazine/pistol/g22, SLOT_IN_ACCESSORY)
-	H.equip_to_slot_or_del(new /obj/item/weapon/gun/pistol/g22(H), SLOT_IN_ACCESSORY)
```
Comment (Lumipharon) [inline code/datums/jobs/job/special_forces.dm:48]: Legitimately why?

File: code/modules/clothing/masks/gasmask.dm [hunk 4]
```diff
@@ -98,6 +98,20 @@
...
+	desc = "A close-fitting tactical mask used by special operation operatives."
+	icon = 'icons/mob/clothing/mask.dmi'
+	item_icons = list(
+		slot_wear_suit_str = 'icons/mob/clothing/mask.dmi'
+	)
+	icon_state = "swat_specops"
+	item_state = "swat_specops"
+	anti_hug = 1
+	siemens_coefficient = 0.7
+	flags_armor_protection = FACE|EYES
```
Comment (Lumipharon) [inline code/modules/clothing/masks/gasmask.dm:112]: don't copy vars blindly, these are inherited.

File: code/modules/clothing/masks/gasmask.dm [hunk 5]
```diff
@@ -98,6 +98,20 @@
 	siemens_coefficient = 0.7
 	flags_armor_protection = FACE|EYES
 
+/obj/item/clothing/mask/gas/swat/mk2
+	name = "\improper SWAT mask MK.2"
+	desc = "A close-fitting tactical mask used by special operation operatives."
+	icon = 'icons/mob/clothing/mask.dmi'
+	item_icons = list(
+		slot_wear_suit_str = 'icons/mob/clothing/mask.dmi'
+	)
```
Comment (Lumipharon) [inline code/modules/clothing/masks/gasmask.dm:107]: Almost certainly ditto with this.

File: code/modules/clothing/suits/marine_armor.dm [hunk 6]
```diff
@@ -697,3 +697,107 @@
 	desc = "A piece of ICCGF body armor, worn by specialized infantry. Most Infantry actions in the ICC forces are done by adhoc personnel due to constant shortages of manpower, however most real Infantry divisions are of high quality, and are better known as 'Guardsmen'.  Protects well from most sources, and will entirely protect from explosions."
 	icon_state = "icc_guard"
 	soft_armor = list(MELEE = 60, BULLET = 65, LASER = 40, ENERGY = 60, BOMB = 85, BIO = 10, FIRE = 55, ACID = 40)
+
+//===========================SPEC OPS================================
+
+/obj/item/clothing/suit/storage/marine/specops
+	name = "\improper Ballistic vest"
```
Comment (Lumipharon) [inline code/modules/clothing/suits/marine_armor.dm:704]: Why's this improper?

File: code/modules/clothing/suits/marine_armor.dm [hunk 7]
```diff
@@ -697,3 +697,107 @@
...
+/obj/item/clothing/suit/storage/marine/specops/medic
+	name = "\improper Ballistic vest"
+	desc = "Civilian type armor, made to combat both melee and projectiles."
+	icon_state = "specops_vest_medic"
+
+/obj/item/clothing/suit/storage/marine/specops/leader
+	name = "\improper Ballistic vest"
+	desc = "Civilian type armor, made to combat both melee and projectiles."
+	icon_state = "specops_vest_leader"
+	slowdown = SLOWDOWN_ARMOR_MEDIUM
```
Comment (Lumipharon) [inline code/modules/clothing/suits/marine_armor.dm:730]: It's literally the same as the normal armour, only a straight downgrade due to a higher slowdown?

File: code/modules/clothing/suits/marine_armor.dm [hunk 8]
```diff
@@ -697,3 +697,107 @@
...
+
+//===========================MARINE VERSION================================
+
+/obj/item/clothing/suit/storage/marine/ballistic
+	name = "\improper Crasher multi-threat ballistic armor"
```
Comment (Lumipharon) [inline code/modules/clothing/suits/marine_armor.dm:735]: Why is this an old armour typepath? Should probably be modular.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/modules/clothing/suits/marine_armor.dm (hunk 7 above)
Comment (SomrNetwork) [reply code/modules/clothing/suits/marine_armor.dm:731]: added new armor stats

File: code/modules/clothing/suits/marine_armor.dm (hunk 6 above)
Comment (SomrNetwork) [reply code/modules/clothing/suits/marine_armor.dm]: removed improper

File: code/datums/jobs/job/special_forces.dm (hunk 3 above)
Comment (SomrNetwork) [reply code/datums/jobs/job/special_forces.dm]: added g22 back

**Comment by Lumipharon** [comment]
stop self resolving feedback.

**Comment by SomrNetwork** [comment]
last update will happen when kuro responds
image

**Comment by SomrNetwork** [comment]
should be complete now, I've already tried fixing the merge conflicts with the new files

**Comment by lbnesquik** [comment]
I'm all for it.

**Comment by SomrNetwork** [comment]
i'm in pain

**Comment by aarrisu** [comment]
wyci

**Comment by SomrNetwork** [comment]
okay ONE last

**Comment by SomrNetwork** [comment]
image

File: code/__DEFINES/loadout_variants.dm [hunk 9]
```diff
@@ -28,6 +28,10 @@
 #define CAPE_KILT_ALT "cape_kilt_alt"
 #define SOM_BLACK "SOM_black"
 #define CAPE_HIGHLIGHT_NORMAL_ALT "normal_alt"
+#define URBAN "ballistic_vest_urban"
+#define DESERT "ballistic_vest_desert"
+#define JUNGLE "ballistic_vest_jungle"
+#define SNOW "ballistic_vest_snow"
```
Comment (Lumipharon) [inline code/__DEFINES/loadout_variants.dm:34]: bro, look at the rest of the defines above, these are not right lol

File: code/__DEFINES/loadout_variants.dm [hunk 10]
```diff
@@ -80,4 +84,8 @@ GLOBAL_LIST_INIT(loadout_variant_keys, list(
 	MARK_ONE_WEBBING = "webbing",
 	FOUR_EYE_FACEPLATE = "Four Eye",
 	FOUR_EYE_FACEPLATE_VISOR = "Four Eye",
+	URBAN = "urban",
+	JUNGLE = "jungle",
+	DESERT = "desert",
+	SNOW = "snow",
```
Comment (Lumipharon) [inline code/__DEFINES/loadout_variants.dm:90]: same as above.

File: code/modules/clothing/modular_armor/xenonauten.dm [hunk 11]
```diff
@@ -178,6 +178,47 @@
...
+		/obj/item/binoculars,
+		/obj/item/weapon/combat_knife,
+		/obj/item/attachable/bayonetknife,
+		/obj/item/storage/belt/sparepouch,
+		/obj/item/storage/holster/blade,
+		/obj/item/storage/holster/belt,
+		/obj/item/weapon/energy/sword,
+	)
+
+/obj/item/clothing/suit/storage/modular/xenonauten/ballistic
```
Comment (Lumipharon) [inline code/modules/clothing/modular_armor/xenonauten.dm:200]: what is this typepath?
it should be like /obj/item/clothing/suit/modular/yourthinghere, if you want it to be like a modular armour type, or just /obj/item/clothing/suit/storage/yourthinghere if its going to stay as a suit/storage type suit

File: code/modules/clothing/masks/gasmask.dm [hunk 12]
```diff
@@ -101,6 +101,17 @@
 	siemens_coefficient = 0.7
 	flags_armor_protection = FACE|EYES
 
+/obj/item/clothing/mask/gas/specops
+	name = "Special Operations gasmask"
+	desc = "A close-fitting tactical mask that can be connected to an air supply. Based off an old design of gas masks."
+	icon = 'icons/mob/clothing/mask.dmi'
+	item_icons = list(
+		slot_wear_suit_str = 'icons/mob/clothing/mask.dmi'
```
Comment (Lumipharon) [inline code/modules/clothing/masks/gasmask.dm:109]: is this uh, actually intended?

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/__DEFINES/loadout_variants.dm (hunk 9 above)
Comment (Lumipharon) [reply code/__DEFINES/loadout_variants.dm]: look at the above defines my guy. There is one for each item type, not variant.

**Review by Kuro020 — APPROVED** [review]

**Pull request merged**